### ISSUE TITLE
Logging hook gets stuck when restarting training

### ISSUE THREAD

**Issue by OMalenfantThuot** [atomistic-machine-learning/schnetpack#283]
Hey, it's me again.

I noticed the learning rate was getting stuck to a certain value in the log, but the training was still deemed complete by the code.
After some testing, it only happens after the job was stopped then restarted.
The logging hook and the scheduling hook, no longer agree on the value of the learning rate at that point.
In the last major [update](https://github.com/atomistic-machine-learning/schnetpack/pull/262), in file `src/schnetpack/train/trainer.py`, a `self.` was removed in line 114. Putting it back solves the problem, but I haven't looked in detail why. Not sure if this change had deeeper implications either.

Thanks again!
Olivier

**Comment by OMalenfantThuot** [comment]
Also a minor thing. The spk_run.py script does not have the executable permissions by default and doing a chmod on it registers as a change by git, which is annoying. Maybe it's worth doing it by default?

**Comment by ktschuett** [comment]
Thanks for the report and sorry for the long response time.

This issue should have already be fixed (#243 , #239).

Regarding the chmod thing: I never noticed this problem. Do you install schnepack `python setup.py [install/develop]` or `pip install . [-e]`? Then the script should be directly in you path and executable.

**Comment by OMalenfantThuot** [comment]
Hi @ktschuett, thanks for the answer.
I didn't know the scripts were added to the path, so ignore the second comment. This works well.
As for the main comment. The update [#243](https://github.com/atomistic-machine-learning/schnetpack/pull/243) is actually the source of the problem. I have to revert it to make it work correctly.

**Comment by ktschuett** [comment]
Ah, ok. We'll have another look

**Comment by Stefaanhess** [comment]
hey @OMalenfantThuot ,
i think i found the bug. Have you been using the `ReduceOnPlateau`-Hook? The `state_dict` for this hook had been saved in a wrong way. #289 should fix it

**Comment by OMalenfantThuot** [comment]
Yes it was the ` ReduceOnPlateau` hook for me. Thanks a lot!

**Comment by Stefaanhess** [comment]
I messed things up with git... #290 should be working :)

**Comment by ktschuett** [comment]
Fixed with #290

**Comment by sirmarcel** [comment]
Yikes, thanks for filing and fixing this, I've been wondering why my training runs have started looking so weird!

**Comment by sirmarcel** [comment]
It seems like I somehow caused this with #243 , but I'm not sure how. Would you mind explaining briefly?

**Comment by Stefaanhess** [comment]
If I remember correctly, before your commit the hooks have not been loaded from the `state_dict` at all (due to the missing `self.`). After you fixed this in #243 the scheduling hooks were loaded from the checkpoint file, but there also was a bug in the `ReduceOnPlateau` state dict, which was not loading the `state_dict` of the scheduler correctly. Does this make sense?

**Comment by Stefaanhess** [comment]
So the #290 bug was hidden behind the #243 bug

**Comment by sirmarcel** [comment]
Okay -- thank you! Time to upgrade. :)
